src/url-params.ts

```diff
@@ -111,6 +111,15 @@ export class AnalyticsUrlParams {
 		this.client = client;
 	}
 
+	/**
+	 *
+	 * @param client
+	 * sets the class client value to whatever client is passed as the function argument
+	 */
+	declareClient(client: Client) {
```

Comment:
How is this related to `getQueryString`?

I get it is related to the initialization of the tracking client but might've been better off as a separate PR. Just nitpicking.

Reply:
I don't think it's nitpicking. I'd also suggest to split such work to a separate PR. We'd need to test the impact of using such a new method a bit further and it could use some more discussion. There is probably no need to hold the original changes here waiting for such further discussion and tests around `declareClient`.

Reply:
My bad, I had added that to be able to test the session_id being passed but was indeed intended for a separate PR as discussed on FD, I'll remove it on the next commit